src/components/Match/Vision/index.jsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+import {
+  Row,
+  Col,
+} from 'react-flexbox-grid';
+import { formatSeconds, getTeamName } from 'utility';
+import Slider from 'material-ui/Slider';
+import _ from 'lodash/fp';
+
+import VisionFilter from './VisionFilter';
+import VisionItems from './VisionItems';
+import VisionMap from './VisionMap' ;
+import VisionLog from './VisionLog';
+import PlayerFilter from './PlayerFilter';
+import styles from './Vision.css';
+
+const SliderTicks = props => (
+  <div className={styles.sliderTicks}>
+    {props.ticks.map((tick) => {
+      const [t, min, max] = [tick, props.min, props.max];
+      const percent = 100 * ((t - min) / (max - min));
+      const cls = [styles.sliderTick];
+
+      if (tick <= props.value) {
+        cls.push(styles.active);
+      }
+
+      return (
+        <a key={tick} onClick={() => props.onTickClick(tick)} className={cls.join(' ')} style={{ left: `${percent}%` }}>
+          {formatSeconds(tick)}
+        </a>
+      );
+    })}
+  </div>
+);
+
+// remove this component
+const PlayersFilter = ({ activeFilters, players, onFilterClick }) => (
+  <div>
+    {players.map((p, index) => <PlayerFilter key={index} player={p} activeFilters={activeFilters} onFilterClick={onFilterClick} />)}
+  </div>
+);
+
+const pipelineFilter = (filters, data) => {
+  const filtered = filters.map(f => data.filter(f))
+                          .reduce((o, v) => o.concat(v), []);
+
+  return _.differenceWith((x, y) => x === y, data, filtered);
+};
+
+const FixedPlayersFilter = PlayersFilter;
+
+class Vision extends React.Component {
+  static hideVisionLog(playerSlot, type) {
+    return l => l.entered.player_slot === playerSlot && l.type === type;
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.sliderMin = -90;
+    this.sliderMax = props.match.duration;
```

Comment:
if you set things based on props in the constructor, you need to do a componentWillReceiveProps check to update the value if the prop value changes.